app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,90 @@
+class RecipesController < ApplicationController
+  # GET /recipes
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.new(recipe_params)
+    respond_to do |f|
+      if @recipe.save
+        f.html { redirect_to my_recipes_path, notice: 'Recipe was successfully created' }
+      else
+        f.html { render :new }
+      end
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id]).destroy
+    respond_to do |f|
+      f.html { redirect_to my_recipes_path, notice: 'Recipe was successfully destroyed' }
+    end
+  end
+
+  def my_recipes
+    @recipes = current_user.recipes.includes(:user)
+  end
+
+  def public_recipes
+    @recipes = Recipe.includes(:user).where(public: true)
+
+    @shopping_lists = {}
+    @recipes.each do |recipe|
+      @shopping_lists[recipe.id] = generate_shopping_list(recipe)
+    end
+  end
+
+  def toggle_public
+    @recipe = Recipe.find(params[:id])
+    @recipe.update(public: !@recipe.public)
+    redirect_to @recipe
+  end
+
+  def shopping_list
+    @recipe = Recipe.find(params[:id])
+    @shopping_list, @total_items, @total_price = generate_shopping_list(@recipe)
+    render 'shopping_lists/index'
+  end
+
+  def generate_shopping_list(recipe)
+    recipe_foods = RecipeFood.where(recipe)
+
+    shopping_list = {}
```

Comment:
- [ ] Again here as requested by the previous reviewer, kindly fix the error that occurs when a user is trying to generate a shopping list
![image](https://github.com/chiztechnology/rails-capstone-project/assets/81932478/1b729d3b-272e-4b0b-a916-48a6c36bc545)
The error is in this method

Reply:
- [ ] This is in line with the requested change by the previous reviewer to kindly fix all your technical functionality to work as expected